lib/linguist/heuristics.rb

```diff
@@ -68,6 +68,20 @@ def call(data)
     # Common heuristics
     ObjectiveCRegex = /^\s*(@(interface|class|protocol|property|end|synchronised|selector|implementation)\b|#import\s+.+\.h[">])/
 
+
+    CPPRegexType0 = /^\s*#.*include.*<(stdint|cstdint|string|vector|map|list|array|bitset|queue|stack|forward_list|unordered_map|unordered_set|(i|o|io)stream).*>/
```

Comment:
Hi @Alhadis, I have applied the  suggested improvements excepting the regex splitting.

The splitting was influencing on tests (breaking them). The first regex related with common CPP includes is a Frankenstein. I tried %r{(...)}x, ~= /.../ and nothing. Maybe it could be something related with my ruby copy but I am not sure.